### ISSUE TITLE
Errors on Vagrantfile are are being masked on WSL

### ISSUE THREAD

**Issue by mmarchini** [hashicorp/vagrant#8609]
### Vagrant version
1.9.5 on both Windows and WSL

### Host operating system
WSL (Windows)

### Guest operating system
Ubuntu 16.04

### Vagrantfile
```ruby
# Add any syntatic or semantic error to your Vagrantfile to reproduce the problem
# In this example I typed agrant instead of Vagrant
agrant.configure(2) do |config|
  config.vm.box = "ubuntu/xenial64"
  config.vm.network "forwarded_port", guest: 8000, host: 8001
  config.vm.synced_folder ".", "/home/ubuntu/project"
  config.vm.provision "shell", path: "install.sh", privileged: false
end
```

### Debug output
https://gist.github.com/mmarchini/a3ea8dde097c6d6b4a5aa33224da5d85

### Expected behavior
A syntax or semantic error should be shown to the user when they exist, but instead a "Version mismatch between Windows and WSL vagrant" error is being displayed.

### Actual behavior
Vagrant should show the actual Vagrantfile error

### Steps to reproduce
1. Edit your vagrant file to add some syntax or semantic error
2. Run some command (for example, `vagrant up`
3. The error that will appear will not reflect the 

I'll try to write a fix for this problem during the weekend.

**Comment by strarsis** [comment]
FWIW, I had the same error message but a different reason.
I ran vagrant in normal Windows cmd (as normal exe in Windows, not in WSL) 
and got an error message about plugins that have to be repaired.
After plugin repair vagrant worked correctly in cmd - and also now in WSL (together with the vagrant installed there).

**Comment by jkeslin-argonne** [comment]
I'm having this same issue on Vagrant 2.0.0 when I've installed both 2.0.0 on windows and on wsl (16.04).  Can someone confirm that installing on both is required for windows 10 CU WSL (16.04)?  I'm seeing confusing installer instructions across the board. 

I've encountered this when running a vagrant up as sudo and then coming back later and trying to vagrant up as my normal user.  Since the uid's don't match, it causes a uid mismatch error, however I don't see the error.. I see the same described message about the two installs not being the same and it appears the windows version is unable to be determined for some reason.  

I fixed my issue by deleting the .vagrant directory and re-running vagrant up as my regular user, however when an error is encountered, it should show the error without having to enable debugging (I can only see the real error when I set the environment variables: 

VAGRANT_LOG="debug"
VAGRANT_DEBUG_LAUNCHER="1"
